pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -159,11 +159,11 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	if checkEngineStatusForStop(engine) {
+	if checkEngineStateForStop(engine) {
 		reconcileResult, err := r.reconcileForDelete(request)
 		return reconcileResult, err
 
-	} else if engine.Instance.Spec.EngineStatus == "start" {
+	} else if engine.Instance.Spec.EngineState == "active" || engine.Instance.Spec.EngineState == "" {
```

Comment:
do we need a nil check for cases where the engineState is not defined at all? Also, should it be omitEmpty (then needs nil check)

Reply:
Needs CRD validation (for string) & values pattern matched for Active/Stop